### PR TITLE
Adding Subgraph to Documentation

### PR DESCRIPTION
**Pull request by PaulieB14**
Adding 4 basic pages for initial subgraph documentation and useful links for others to query Sablier's subgraph. Provided entity description and 2 sample queries.

### CODE REVIEW

**Comment by maxdesalle** [comment]
Thank you @PaulieB14 @Data-Nexus, really appreciate it! One minor note: for future pull requests, please limit the number of git commits and add descriptive commit messages so that others can quickly figure what was exactly done in that specific commit.

Anyway, great work! Added in [484b24f](https://github.com/sablierhq/docs/commit/4b85a33024626ceb037d339bfb2b2efbfb9bbdc4).

File: protocol/Subgraph/Queries.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+Below are some sample queries you can use to gather information from the Sablier contracts. 
+
+You can build your own queries using a [GraphQL Explorer](https://graphiql-online.com/graphiql) and enter your endpoint to limit the data to exactly what you need.
+
+### Sender Streams
+
+Description: Gathers the streams created by an account as well as the stream's most recent withdrawal
+
+```
```
Comment (PaulRBerg) [inline protocol/Subgraph/Queries.md]: Could we specify the language here? Like this:

```text
```graphql
```

File: protocol/Subgraph/Queries.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  }
+}
+
+```
+
+### Receiver Streams
+
+Description: Gathers the streams an account receives as well as the stream's most recent withdrawal
+
+```
```
Comment (PaulRBerg) [inline protocol/Subgraph/Queries.md]: Ditto as above:

```text
```graphql
```

File: protocol/Subgraph/SubgraphData.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+This subgraph is can be used to query Sablier data.
+
+Subgraph information is serviced by a decentralized group of server operators called Indexers.
+
+## Ethereum Mainnet
+
+[Creating an API Key Video Tutorial](https://www.youtube.com/watch?v=UrfIpm-Vlgs)
+
+- [Explorer Page](https://thegraph.com/explorer/subgraph?id=A5XzhYZ4muyRegVTYmwWbCbKWvD4LTWqac43CEGTEGdK&view=Playground)
+- Graphql Endpoint: https://gateway.thegraph.com/api/[api-key]/subgraphs/id/A5XzhYZ4muyRegVTYmwWbCbKWvD4LTWqac43CEGTEGdK
```
Comment (PaulRBerg) [inline protocol/Subgraph/SubgraphData.md]: Can we write this as "GraphQL", and turn it into a hyperlink?

File: protocol/Subgraph/SubgraphData.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+- [Code Repo](https://github.com/sablierhq/subgraph)
```
Comment (PaulRBerg) [inline protocol/Subgraph/SubgraphData.md]: Could we refer to this as "Subgraph Source Code"?

File: protocol/Subgraph/SubgraphData.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---
+sidebar_position: 1
+title: Subgraph Data Introduction
+---
+
+# Sablier Subgraph Introduction
+
+Sablier has a GraphQL API Endpoint hosted by [The Graph](https://thegraph.com/docs/about/introduction#what-the-graph-is) called a subgraph for indexing and organizing data from the Sablier smart contracts.
+
+This subgraph is can be used to query Sablier data.
```
Comment (PaulRBerg) [inline protocol/Subgraph/SubgraphData.md]: Typo: extra "is".

File: protocol/Subgraph/Queries.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      amount
+      timestamp
+      txhash
+    }
+  }
+}
+```
+
+
+```
```
Comment (PaulRBerg) [inline protocol/Subgraph/Queries.md]: Stranded triple backticks. Should be deleted.

File: protocol/Subgraph/AdditionalResources.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+---
+sidebar_position: 4
+title: Helpful Links
+---
+
+## Helpful Links
+
+[Querying from an Application](https://thegraph.com/docs/en/developer/querying-from-your-app/) 
+
+[Managing your API Key & Setting your indexer preferences](https://thegraph.com/docs/en/studio/managing-api-keys/ )
```
Comment (PaulRBerg) [inline protocol/Subgraph/AdditionalResources.md]: Can re write this as an unordered list (with bullets)?

**Review by PaulRBerg — CHANGES_REQUESTED** [review]
Thanks for your contribution @PaulieB14, @Data-Nexus. We're happy to accept this PR, but before we do there are a few things we need to fix. I've left specific remarks via the GitHub Review flow, there's two other changes I wish to request:

- [ ] For consistency purposes, could you please rename the file names to use `kebab-case` instead of `CapitalCase`?
- [ ] Could you install the [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) extension in VSCode, enable the [`formatOnSave`](https://stackoverflow.com/questions/39494277/how-do-you-format-code-on-save-in-vs-code) setting, and run <kbd>CMD</kbd>+<kbd>S</kbd> in the `Entities.md` file? This would auto-format the Markdown tables according to our Prettier rules.

Oh, and as @maxdesalle has said, it would be very helpful for brevity purposes to make fewer commits and provide more specific commit messages. Not a biggy for us at this time, since we can just squash the commit messages before merging the PR, but this is good to know FYI!

File: protocol/Subgraph/SubgraphData.md (hunk 3 above)
Comment (Data-Nexus) [reply protocol/Subgraph/SubgraphData.md]: Only reason why I would recommend against a hyperlink is the API key still needs to be input before it's a usable endpoint.

Thoughts?

Comment (PaulRBerg) [reply protocol/Subgraph/SubgraphData.md]: Oh, I see. In this case, I would recommend explaining this tidbit of information explicitly in a sentence. See how I refactored this in my latest commit.

**Pull request closed without merging**